process/headerCheck/headerSignatureVerify.go

```diff
@@ -370,6 +365,10 @@ func (hsv *HeaderSigVerifier) VerifyHeaderProof(proofHandler data.HeaderProofHan
 		return fmt.Errorf("%w for flag %s", process.ErrFlagNotActive, common.EquivalentMessagesFlag)
 	}
 
+	if proofHandler.GetIsStartOfEpoch() {
+		return hsv.verifyHeaderProofAtTransition(proofHandler)
```

Comment:
do we need to trigger this flow if previous epoch is with equivalent proofs, and the transition is also to an epoch with equivalent proofs?

Reply:
maybe IsStartOfEpoch and previous epoch not with proofs?

Reply:
we can also have an unit test case for this scenario

Reply:
very good catch, updated in 2 places